topk-py/src/client/async/collection.rs

```diff
@@ -0,0 +1,139 @@
+use crate::data::value::{NativeValue, Value};
+use crate::error::RustError;
+use crate::query::{ConsistencyLevel, Query};
+use pyo3::prelude::*;
+use pyo3_async_runtimes::tokio::future_into_py;
+use std::{collections::HashMap, sync::Arc};
+
+#[pyclass]
+pub struct AsyncCollectionClient {
+    client: Arc<topk_rs::Client>,
+    collection: String,
+}
+
+impl AsyncCollectionClient {
+    pub fn new(client: Arc<topk_rs::Client>, collection: String) -> Self {
+        Self { client, collection }
+    }
+}
+
+#[pymethods]
+impl AsyncCollectionClient {
+    #[pyo3(signature = (ids, fields=None, lsn=None, consistency=None))]
+    pub fn get(
+        &self,
+        py: Python<'_>,
+        ids: Vec<String>,
+        fields: Option<Vec<String>>,
+        lsn: Option<String>,
+        consistency: Option<ConsistencyLevel>,
+    ) -> PyResult<PyObject> {
+        let client = self.client.clone();
+        let collection = self.collection.clone();
+
+        future_into_py(py, async move {
+            let docs = client
+                .collection(&collection)
+                .get(ids, fields, lsn, consistency.map(|c| c.into()))
+                .await
+                .map_err(RustError)?;
+
+            Ok(docs
```

Comment:
nit: having `Document` here would be nicer

Reply:
added here https://github.com/topk-io/topk/pull/233/commits/fa0ace7ef1736c37ed3ec3cc1f16be5c6415b08a